src/merkle.rs

```diff
@@ -1,51 +1,182 @@
-use ff::{PrimeField, BitIterator};
+use std::ops::{Deref, DerefMut};
+use ff::{PrimeField, BitIterator, Field};
 use pairing::bls12_381::Fr;
 use zcash_primitives::jubjub::JubjubEngine;
 use zcash_primitives::pedersen_hash;
+use crate::Params;
 
-#[derive(Debug, Clone, Copy, Eq, PartialEq)]
-pub enum PathDirection {
-    Left,
-    Right,
+/// A point in the authentication path.
+#[derive(Clone, Debug)]
+pub struct AuthPathPoint<E: JubjubEngine> {
+    /// The current selection. That is, the opposite of sibling.
+    pub current_selection: MerkleSelection,
+    /// Sibling value, if it exists.
+    pub sibling: Option<E::Fr>,
 }
```

Comment:
It's worth considering if this formulation gives cleaner code elsewhere or if we'd do better with say
```
pub enum MerkleStep<E: JubjubEngine> {
    Left(pub Option<E::Fr>),
    Right(pub Option<E::Fr>)
}
```

Reply:
Will do. Indeed that looks cleaner now.

But just want to note that in the future I do think this can be further improved to avoid the alignment issue -- the "left" or "right" can be stored in a single structure in the end, or we can use something like [generalized merkle index](https://github.com/ethereum/eth2.0-specs/blob/2787fea5feb8d5977ebee7c578c5d835cff6dc21/specs/light_client/merkle_proofs.md#generalized-merkle-tree-index).